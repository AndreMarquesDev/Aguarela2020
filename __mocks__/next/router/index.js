jest.mock('next/router', () => ({
    ...jest.requireActual('next/router'),
    useRouter() {
        return {
            route: '/',
            pathname: '',
            query: {
                language: 'pt',
            },
            asPath: '',
        };
    },
}));

module.exports = jest.requireMock('next/router');
