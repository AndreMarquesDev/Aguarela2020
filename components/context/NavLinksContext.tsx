import { createContext, Context } from 'react';

export interface NavLinksContextProps {
    setNavHeight?: React.Dispatch<React.SetStateAction<number>>;
}

const NavLinksContext: Context<NavLinksContextProps> = createContext(null);

export default NavLinksContext;
