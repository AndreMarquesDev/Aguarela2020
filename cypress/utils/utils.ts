import { Locale } from '../../types/Locale';
import {
    menuIconButtonDataTestId,
    nukaCarouselNextButtonDataTestId,
} from '../../utils/dataTestIds';
import { TextsInterface, textsPt, textsEn } from '../../utils/texts';

export const getLocalizedTexts = (locale: Locale): TextsInterface => {
    if (locale === Locale.Pt) {
        return textsPt;
    }

    return textsEn;
};

export const openMenuMobile = (isMobile: boolean, isFirefox: boolean): void => {
    if (isMobile) {
        cy.getByDataTestId(menuIconButtonDataTestId).click({ force: isFirefox });
    }
};

export const clickNextArrowButtonIfMobile = (
    isMobile: boolean,
    container: string,
    containerTitle: string
): void => {
    const isFirefox = Cypress.isBrowser('firefox');

    if (isMobile) {
        cy.getByDataTestIdParent(container, nukaCarouselNextButtonDataTestId).click({
            force: isFirefox,
        });

        cy.getByText(container, containerTitle).scrollIntoView();
    }
};
