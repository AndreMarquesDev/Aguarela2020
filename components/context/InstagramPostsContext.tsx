import { createContext, Context } from 'react';
import { IInstagramPost } from '../../utils/generic';

export interface IInstagramPostsContext {
    posts: IInstagramPost[];
}

const InstagramPostsContext: Context<IInstagramPostsContext> = createContext(null);

export default InstagramPostsContext;
