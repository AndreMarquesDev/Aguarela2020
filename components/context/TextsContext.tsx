import { createContext, Context } from 'react';
import { TextsInterface, textsPt } from '../../utils/texts';

export interface TextsContextProps {
    texts: TextsInterface;
}

export const TextsContext: Context<TextsContextProps> = createContext({
    texts: textsPt,
});
