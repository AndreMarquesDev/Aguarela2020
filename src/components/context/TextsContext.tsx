import type { Context } from 'react';
import type { TextsInterface } from '../../utils/texts';
import { createContext } from 'react';
import { textsPt } from '../../utils/texts';

export type TextsContextProps = {
    texts: TextsInterface;
};

export const TextsContext: Context<TextsContextProps> = createContext({
    texts: textsPt,
});
