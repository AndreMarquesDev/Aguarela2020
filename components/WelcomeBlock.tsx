import React, { FC, useContext } from 'react';
import TextsContext from './context/TextsContext';

const WelcomeBlock: FC = () => {
    const { texts } = useContext(TextsContext);

    return (
        <>
            <section className="container">
                <div className="wrapper genericMargins">
                    <div className="titleBlock">
                        <strong>{texts.welcome1}</strong>
                        <strong>{texts.welcome2}</strong>
                        <strong>{texts.welcome3}</strong>
                    </div>
                    <div className="textBlock">
                        <p
                            // eslint-disable-next-line react/no-danger
                            dangerouslySetInnerHTML={{
                                __html: texts.managingsocialMedia,
                            }}
                            className="bodyText bodyText--white"
                        />
                        <p className="bodyText bodyText--white">{texts.monitorAndOptimizeProcessAndStrategy}</p>
                        <p className="bodyText bodyText--white">{texts.throughStrategicPlanning}</p>
                    </div>
                </div>
            </section>

            <style jsx>
                {`
                    @import './styles/_vars.scss';

                    .container {
                        background: $purple;

                        .wrapper {
                            display: flex;
                            justify-content: space-between;
                            align-items: center;

                            @include desktop {
                                flex-direction: column;
                                justify-content: center;
                            }
                        }

                        .titleBlock {
                            width: 35%;

                            @include desktop {
                                width: 100%;
                                margin-bottom: 50rem;
                            }

                            strong {
                                display: block;
                                @include fontXXL($textTransform: uppercase, $fontWeight: 900);
                                letter-spacing: 30rem;
                                text-align: right;
                                -webkit-filter: drop-shadow(20rem -10rem 0 $pink);
                                filter: drop-shadow(20rem -10rem 0 $pink);
                                color: $blue;
                                background: $blue;
                                -webkit-background-clip: text;
                                -webkit-text-fill-color: transparent;

                                @include desktop {
                                    text-align: center;
                                }

                                @include mobile {
                                    @include fontXL($textTransform: uppercase, $fontWeight: 900);
                                    -webkit-filter: drop-shadow(10rem -5rem 0 $pink);
                                    filter: drop-shadow(10rem -5rem 0 $pink);
                                }
                            }
                        }

                        .textBlock {
                            width: 60%;

                            @include desktop {
                                width: 80%;
                            }

                            @include tablet {
                                width: 100%;
                            }

                            p {
                                @include mobile {
                                    @include fontS($white);
                                }
                            }
                        }
                    }
                `}
            </style>
        </>
    );
};

export default WelcomeBlock;
