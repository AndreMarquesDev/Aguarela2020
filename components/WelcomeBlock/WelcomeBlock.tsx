import { useRouter } from 'next/router';
import React, { useContext } from 'react';
import { welcomeSectionDataTestId } from '../../utils/dataTestIds';
import { Locale } from '../../types/Locale';
import { TextsContext } from '../context/TextsContext';
import { TextShadowBlock } from '../TextShadowBlock/TextShadowBlock';

export const WelcomeBlock = (): JSX.Element => {
    const { texts } = useContext(TextsContext);

    const { query } = useRouter();
    const currentLanguage = query.language?.toString() as Locale;
    const titleMobile =
        currentLanguage === Locale.Pt
            ? `${texts.welcome1}-${texts.welcome2}${texts.welcome3}`
            : `${texts.welcome1}${texts.welcome2}${texts.welcome3}`;

    return (
        <TextShadowBlock
            dataTestId={welcomeSectionDataTestId}
            textBlock1={texts.managingsocialMedia}
            textBlock2={texts.monitorAndOptimizeProcessAndStrategy}
            textBlock3={texts.throughStrategicPlanning}
            title1={texts.welcome1}
            title2={texts.welcome2}
            title3={texts.welcome3}
            titleMobile={titleMobile}
        />
    );
};
