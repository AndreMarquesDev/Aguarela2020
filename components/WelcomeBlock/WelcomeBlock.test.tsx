import '@testing-library/jest-dom';
import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';
import * as nextRouter from 'next/router';
import { WelcomeBlock } from './WelcomeBlock';
import { textsPt, textsEn } from '../../utils/texts';
import { TextsContext } from '../context/TextsContext';
import { setJestWindowWidth } from '../../utils/jest/setJestWindowWidth';
import { Breakpoint } from '../../utils/useWindowSize';
import { textBlock1DataTestId } from '../../utils/dataTestIds';

// @ts-ignore
nextRouter.useRouter = jest.fn(() => ({
    query: {
        language: 'pt',
    },
}));

const renderComponent = (): RenderResult => {
    return render(<WelcomeBlock />);
};

describe('<WelcomeBlock />', () => {
    test('renders properly', () => {
        const { container } = renderComponent();

        const title1 = screen.getByText(textsPt.welcome1);
        const title2 = screen.getByText(textsPt.welcome2);
        const title3 = screen.getByText(textsPt.welcome3);
        const textBlock2 = screen.getByText(textsPt.monitorAndOptimizeProcessAndStrategy);
        const textBlock3 = screen.getByText(textsPt.throughStrategicPlanning);

        const textBlock1 = screen.getByTestId(textBlock1DataTestId);
        const actualTextBlock1Text =
            'Gestão de redes sociais está no meu ADN. Sou uma freelancer, baseada em Lisboa, com experiência em gestão de redes sociais e criação de conteúdo. A @Aguarela surgiu com o objetivo de ajudar negócios locais e pequenas empresas a posicionarem-se no digital, gerar awareness, criar uma relação com o target e impulsionar vendas';

        expect(title1).toBeInTheDocument();
        expect(title2).toBeInTheDocument();
        expect(title3).toBeInTheDocument();
        expect(textBlock2).toBeInTheDocument();
        expect(textBlock3).toBeInTheDocument();

        expect(textBlock1).toHaveTextContent(actualTextBlock1Text);

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        const { container } = render(
            <TextsContext.Provider
                value={{
                    texts: textsEn,
                }}
            >
                <WelcomeBlock />
            </TextsContext.Provider>
        );

        const title1 = screen.getByText(textsEn.welcome1);
        const title2 = screen.getByText(textsEn.welcome2);
        const title3 = screen.getByText(textsEn.welcome3);
        const textBlock2 = screen.getByText(textsEn.monitorAndOptimizeProcessAndStrategy);
        const textBlock3 = screen.getByText(textsEn.throughStrategicPlanning);

        const textBlock1 = screen.getByTestId(textBlock1DataTestId);
        const actualTextBlock1Text =
            "Social media management is in my DNA. I'm a freelancer, based in Lisbon, with experience in social media management and content creation. @Aguarela's goal is to help local businesses and small companies position themselves digitally, create awareness, create a realtionship with the target and boost sales";

        expect(title1).toBeInTheDocument();
        expect(title2).toBeInTheDocument();
        expect(title3).toBeInTheDocument();
        expect(textBlock2).toBeInTheDocument();
        expect(textBlock3).toBeInTheDocument();

        expect(textBlock1).toHaveTextContent(actualTextBlock1Text);

        expect(container).toMatchSnapshot();
    });

    test('renders properly on mobile', () => {
        setJestWindowWidth(Breakpoint.Mobile);

        renderComponent();

        const titleMobile = screen.getByText(
            `${textsPt.welcome1}-${textsPt.welcome2}${textsPt.welcome3}`
        );

        expect(titleMobile).toBeInTheDocument();
    });

    test('renders properly on mobile in English', () => {
        (nextRouter.useRouter as jest.Mock).mockImplementationOnce(() => ({
            query: {
                language: 'en',
            },
        }));

        setJestWindowWidth(Breakpoint.Mobile);

        render(
            <TextsContext.Provider
                value={{
                    texts: textsEn,
                }}
            >
                <WelcomeBlock />
            </TextsContext.Provider>
        );

        const titleMobile = screen.getByText(
            `${textsEn.welcome1}${textsEn.welcome2}${textsEn.welcome3}`
        );

        expect(titleMobile).toBeInTheDocument();
    });
});
