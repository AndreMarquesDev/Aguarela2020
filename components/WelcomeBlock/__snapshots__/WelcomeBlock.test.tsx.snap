// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeBlock /> renders properly 1`] = `
<div>
  <section
    class="jsx-2290377079 container"
  >
    <div
      class="jsx-2290377079 wrapper genericMargins"
    >
      <div
        class="jsx-2290377079 titleBlock"
      >
        <strong
          class="jsx-2290377079"
        >
          Bem
        </strong>
        <strong
          class="jsx-2290377079"
        >
          Vin
        </strong>
        <strong
          class="jsx-2290377079"
        >
          Dos
        </strong>
      </div>
      <div
        class="jsx-2290377079 textBlock"
      >
        <p
          class="jsx-2290377079 bodyText bodyText--white"
          data-testid="textShadowBlock_textBlock1"
        >
          Gestão de redes sociais está no meu ADN. Sou uma freelancer, baseada em Lisboa, com experiência em gestão de redes sociais e criação de conteúdo. A 
          <a
            class="link"
            href="https://www.instagram.com/aguareladigital"
            rel="noreferrer"
            target="_blank"
          >
            @Aguarela
          </a>
           surgiu com o objetivo de ajudar negócios locais e pequenas empresas a posicionarem-se no digital, gerar awareness, criar uma relação com o target e impulsionar vendas.
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Do ínicio ao fim, acompanho, monitorizo e otimizo todo o processo e estratégia de gestão de redes sociais.
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Através do planeamento estratégico, criação de conteúdo e implementação estou disponível para ser o vosso braço direito.
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`<WelcomeBlock /> renders properly in English 1`] = `
<div>
  <section
    class="jsx-2290377079 container"
  >
    <div
      class="jsx-2290377079 wrapper genericMargins"
    >
      <div
        class="jsx-2290377079 titleBlock"
      >
        <strong
          class="jsx-2290377079"
        >
          Wel
        </strong>
        <strong
          class="jsx-2290377079"
        >
          Co
        </strong>
        <strong
          class="jsx-2290377079"
        >
          Me
        </strong>
      </div>
      <div
        class="jsx-2290377079 textBlock"
      >
        <p
          class="jsx-2290377079 bodyText bodyText--white"
          data-testid="textShadowBlock_textBlock1"
        >
          Social media management is in my DNA. I'm a freelancer, based in Lisbon, with experience in social media management and content creation. 
          <a
            class="link"
            href="https://www.instagram.com/aguareladigital"
            rel="noreferrer"
            target="_blank"
          >
            @Aguarela
          </a>
          's goal is to help local businesses and small companies position themselves digitally, create awareness, create a realtionship with the target and boost sales.
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          From start to finish I keep up with, monitor and optimize the whole social media management process and strategy.
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Through strategic planning, content creation and implementation I can be your right hand.
        </p>
      </div>
    </div>
  </section>
</div>
`;
